### PR TITLE
FreeBSD supports ppoll(2)

### PR DESCRIPTION
**Pull request by gahr**
This also fixes this test error `SubprocessTest.SetWithLotsninja: fatal: pipe: Too many open files` caused by 
```cpp
if (fd_ >= static_cast<int>(FD_SETSIZE))                                      
    Fatal("pipe: %s", strerror(EMFILE)); 
```
in Subprocess::Start

### CODE REVIEW

**Comment by MagikBSD** [comment]
ppoll() is available only on FreeBSD 10.2 and up, but not on older, but still supported releases.
EOL for 9.3 and 10.1 are established to 31 Dec 2016.

**Comment by gahr** [comment]
You are right. Let's keep this here until we EOL older releases.

**Comment by evmar** [comment]
Is there a way to test the FreeBSD version at compile-time?  (E.g. an ifdef?)

**Comment by MagikBSD** [comment]
#include <osreldate.h>
#if (__FreeBSD_version >= 1002000)

**Comment by gahr** [comment]
It's in sys/pram.h

**Comment by gahr** [comment]
... sys/param.h

File: src/subprocess-posix.cc [hunk 1]
```diff
@@ -24,6 +24,13 @@
 #include <sys/wait.h>
 #include <spawn.h>
 
+#ifdef __FreeBSD__
+#  include <sys/param.h>
+#  if defined USE_PPOLL && __FreeBSD_version < 1002000
```
Comment (nico) [inline src/subprocess-posix.cc]: fwiw we don't put spaces between # and the word after it in ninja; maybe you can zap them in a follow-up

**Comment by gahr** [comment]
Done in #1187

**Pull request merged**